module/zfs/arc.c

```diff
@@ -3935,8 +3952,10 @@ arc_evict_hdr(arc_buf_hdr_t *hdr, kmutex_t *hash_lock)
 			ARCSTAT_BUMP(arcstat_mutex_miss);
 			break;
 		}
-		if (buf->b_data != NULL)
+		if (buf->b_data != NULL) {
```

Comment:
This is a comment for line #3945 (but I can't add a comment there since that line was not changed):

For completeness (NIT), I would suggest adding `real_evicted += 0;` so that it's clear that we didn't miss a case to update the real_evicted valued.

Reply:
Since at line 3945 clearly nothing is evicted, I don't see what can be made more clear there.